pkg/repo/chartrepo.go

```diff
@@ -114,31 +108,37 @@ func (r *ChartRepository) Load() error {
 }
 
 // DownloadIndexFile fetches the index from a repository.
-func (r *ChartRepository) DownloadIndexFile() error {
+func (r *ChartRepository) DownloadIndexFile() (string, error) {
 	var indexURL string
 	parsedURL, err := url.Parse(r.Config.URL)
 	if err != nil {
-		return err
+		return "", err
 	}
 	parsedURL.Path = strings.TrimSuffix(parsedURL.Path, "/") + "/index.yaml"
 
 	indexURL = parsedURL.String()
 	// TODO add user-agent
-	resp, err := r.Client.Get(indexURL)
+	resp, err := r.Client.Get(indexURL,
+		getter.WithURL(r.Config.URL),
+		getter.WithTLSClientConfig(r.Config.CertFile, r.Config.KeyFile, r.Config.CAFile),
+		getter.WithBasicAuth(r.Config.Username, r.Config.Password),
+	)
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	index, err := ioutil.ReadAll(resp)
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	if _, err := loadIndex(index); err != nil {
-		return err
+		return "", err
 	}
 
-	return ioutil.WriteFile(helmpath.CacheIndex(r.Config.Name), index, 0644)
+	fname := filepath.Join(r.CachePath, r.Config.Name+"-index.yaml")
```

Comment:
Should this be some functionality inside of `helmpath` or is this just a one off?